Src/VimCore/EditorUtil.fs

```diff
@@ -254,49 +296,140 @@ type SnapshotLineRange  =
+    // table of physical positions corresponding to logical columns
+    new (point: SnapshotPoint) =
+        let line = point.GetContainingLine()
+        let mutable currentPoint = line.Start
+        let endPoint = line.EndIncludingLineBreak
 
-    member x.Subtract count = 
+        // Build an array of positions (including an extra final position)
+        // that we can use to translate from column to position.
+        let positions =
+            seq {
+                while currentPoint.Position < endPoint.Position do
+                    yield currentPoint.Position
+                    let span =EditorCoreUtil.GetCharacterSpan line currentPoint
+                    currentPoint <- span.End
+                yield endPoint.Position
+            }
+            |> Seq.toArray
+
+        // Scan the positions array for the column corresponding to point.
+        let position = point.Position
+        let columnNumber =
+            let mutable columnNumber = positions.Length - 1
+            for i = 0 to positions.Length - 2 do
+                if position >= positions.[i] && position < positions.[i + 1] then
+                    columnNumber <- i
+            columnNumber
+
+        // Once determined, the snapshot line and the position table
+        // are reused for other columns within the line.
+        { _line = line; _positions = positions; _columnNumber = columnNumber }
+
+    /// Constructor for a new column from the same line as another column
+    /// (Warning: requires that you have validated the column number)
+    new (column: SnapshotColumn, columnNumber: int) =
+        Contract.Requires(columnNumber >= 0 && columnNumber <= column.ColumnCount)
+        { _line = column._line; _positions = column._positions; _columnNumber = columnNumber }
+
+    /// The snapshot line containing the column
+    member x.Line = x._line
+
+    /// The column number of the column
+    /// (Warning: don't use the column number as a buffer position offset)
+    /// TODO: Rename this to "ColumnNumber" analogous to LineNumber.
+    member x.Column = x._columnNumber
+
+    /// The position or text buffer offset of the column
+    member x.Position = x._positions.[x._columnNumber]
+
+    /// The point where the column begins
+    member x.Point = new SnapshotPoint(x.Snapshot, x.Position)
+
+    /// The position offset of the column relative the beginning of the line
+    member x.Offset = x.Position - x._positions.[0]
+
+    /// The width of the column in terms of position
+    member x.Width = x._positions.[x._columnNumber + 1] - x.Position
+
+    /// The snapshot span covering the logical character at the column
+    member x.Span = new SnapshotSpan(x.Snapshot, x.Position, x.Width)
+
+    // The number of columns in the line containing the column
+    member x.ColumnCount = x._positions.Length - 1
+
+    /// Whether the "character" at column is a linebreak
+    member x.IsLineBreak = x.Position = x._line.End.Position

```

Comment:
Why this vs. IsInsideLineBreak?

Reply:
With the new snapshot column, a linebreak is a single column position, and so you are either at the linebreak or you aren't. In other words, it's impossible for the snapshot column to be positioned in the middle of a linebreak (or in the middle of a surrogate pair, for that matter).

Reply:
Now reverted.

Reply:
To clarify, the old `SnapshotColumn` is back and unchanged. The new `SnapshotCharacterSpan` still has an `IsLineBreak` property.